contracts/script/setup.ts

```diff
@@ -1,219 +1,349 @@
+  l1Port = 0,
+  l2Port = 0,
+  urgPort = 0,
+  numAccounts = 5,
+  mnemonic = "test test test test test test test test test test test junk",
+  saveDeployments = false,
+}: {
+  l1ChainId?: number;
+  l2ChainId?: number;
+  l1Port?: number;
+  l2Port?: number;
+  urgPort?: number;
+  numAccounts?: number;
+  mnemonic?: string;
+  saveDeployments?: boolean;
+} = {}) {
+  console.log("Deploying ENSv2...");
 
-  const l2Deploy = await executeDeployScripts(
-    resolveConfig(
-      await readConfig({
-        askBeforeProceeding: false,
-        network: "l2-local",
-      }),
-    ),
+  const l1Anvil = anvil({
+    chainId: l1ChainId,
+    port: l1Port,
+    accounts: numAccounts,
+    mnemonic,
+  });
+  const l2Anvil = anvil({
+    chainId: l2ChainId,
+    port: l2Port,
+    accounts: numAccounts,
+    mnemonic,
+  });
+
+  // use same accounts on both chains
+  const accounts = Array.from({ length: numAccounts }, (_, i) =>
+    Object.assign(mnemonicToAccount(mnemonic, { addressIndex: i }), {
+      name: `unnamed${i}`, // default name
+    }),
   );
 
-  console.log("Cross-chain environment setup complete!");
+  // name accounts (exposed as `namedAccounts` in rocketh)
+  const deployer = accounts[0];
+  deployer.name = "deployer";
+  accounts[1].name = "owner";
 
-  const account = mnemonicToAccount(
-    "test test test test test test test test test test test junk",
-  );
+  // shutdown functions for partial initialization
+  const finalizers: (() => Promise<void>)[] = [];
+  async function shutdown() {
+    await Promise.allSettled(finalizers.map((f) => f()));
+  }
 
-  const l1Chain = {
-    id: l1Config.chainId,
-    name: 'L1 Local',
-    nativeCurrency: { name: 'Ether', symbol: 'ETH', decimals: 18 },
-    rpcUrls: {
-      default: { http: [`http://127.0.0.1:${l1.port}`] },
-      public: { http: [`http://127.0.0.1:${l1.port}`] },
-    },
-  } as const;
-
-  const l2Chain = {
-    id: l2Config.chainId,
-    name: 'L2 Local',
-    nativeCurrency: { name: 'Ether', symbol: 'ETH', decimals: 18 },
-    rpcUrls: {
-      default: { http: [`http://127.0.0.1:${l2.port}`] },
-      public: { http: [`http://127.0.0.1:${l2.port}`] },
-    },
-  } as const;
-
-  const l1Client = createClient({
-    chain: l1Chain,
-    transport: webSocket(`ws://127.0.0.1:${l1.port}`, {
-      retryCount: 0,
-    }),
-    account,
-  });
-  const l1Contracts = createDeploymentGetter(l1Deploy, l1Client);
+  try {
+    console.log("Launching L1");
+    await l1Anvil.start();
+    finalizers.push(() => l1Anvil.stop());
+
+    console.log("Launching L2");
+    await l2Anvil.start();
+    finalizers.push(() => l2Anvil.stop());
+
+    // parse `host:port` from the anvil boot message
+    const [l1HostPort, l2HostPort] = [l1Anvil, l2Anvil].map((anvil) => {
+      const message = anvil.messages.get().join("\n").trim();
+      const match = message.match(/Listening on (.*)$/);
+      if (!match) throw new Error(`expected host: ${message}`);
+      return match[1];
+    });
```

Comment:
is this required? prool anvil instance logic already does a `Listening on` check